Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/ApiGatewayResponseExtensions.cs

```diff
@@ -0,0 +1,108 @@
+﻿namespace Amazon.Lambda.TestTool.Extensions;
+
+using Amazon.Lambda.APIGatewayEvents;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using System;
+using System.IO;
+using System.Text;
+
+/// <summary>
+/// Provides extension methods for converting API Gateway responses to HttpResponse objects.
+/// </summary>
+public static class ApiGatewayResponseExtensions
+{
+    /// <summary>
+    /// Converts an APIGatewayProxyResponse to an HttpResponse.
+    /// </summary>
+    /// <param name="apiResponse">The API Gateway proxy response to convert.</param>
+    /// <returns>An HttpResponse object representing the API Gateway response.</returns>
+    public static HttpResponse ToHttpResponse(this APIGatewayProxyResponse apiResponse)
+    {
+        var httpContext = new DefaultHttpContext();
+        var response = httpContext.Response;
+
+        response.StatusCode = apiResponse.StatusCode;
+
+        SetResponseHeaders(response, apiResponse.Headers, apiResponse.MultiValueHeaders);
+
+        SetResponseBody(response, apiResponse.Body, apiResponse.IsBase64Encoded);
+
+        return response;
+    }
+
+    /// <summary>
+    /// Converts an APIGatewayHttpApiV2ProxyResponse to an HttpResponse.
+    /// </summary>
+    /// <param name="apiResponse">The API Gateway HTTP API v2 proxy response to convert.</param>
+    /// <returns>An HttpResponse object representing the API Gateway response.</returns>
+    public static HttpResponse ToHttpResponse(this APIGatewayHttpApiV2ProxyResponse apiResponse)
```

Comment:
so the 6mb part is actually a little tricky. the reason being is that the lambda will return something like

```
{
  "StatusCode": 200,
  "Headers": {
    "Content-Type": "application/json",
    "Custom-Header": "Some Value"
  },
  "Body": "This is the response body",
  "IsBase64Encoded": false
}

```

and then from our end we will have the already deserialized object of APIGatewayHttpApiV2ProxyResponse with that contents. so we need to check the size of the actual serialized APIGatewayHttpApiV2ProxyResponse  object represented in json form right? if thats the case then that gets tricky, because then how do we handle null values?

if we include them theres no way to know if the user sent back

```
Cookies = null
```

in the response (which would be part of the payload size) or if they did not set it (and the payload size would actually be less(\).

Reply:
also from my testing, this size check happens at the lambda layer (at least the 6mb one does) so we may want to have this in another spot

Reply:
I agree we will need to add the check elsewhere.